pkg/health/composite_indicator.go

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package health
+
+// CompositeIndicator aggregates multiple health indicators and provides one detailed health
+type CompositeIndicator struct {
```

Comment:
I am not sure about the composite indicator - if one registers two composite indicators that each has a different aggregator, he should be able to register a top level agregator too, which aggregates the result of the most outer idicators. And this might result in unnecessay complexity

How about we have in the registry just a slice of indicators and one agregation policy? this way the consumer can define everything and it will be simpler